kernel/src/fs/thread_fs.rs

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use ostd::sync::{RwLock, RwMutex};
+
+use super::{fs_resolver::FsResolver, utils::FileCreationMask};
+
+/// FS information for a POSIX thread.
+pub struct ThreadFs {
```

Comment:
I wonder is the name `ThreadFsInfo` more appropriate? It's a little bit longer, but more precise - align with the "filesystem information" wording in the man page, and less ambiguous - typically the name `XXXFs` indicates a specific type of filesystem itself.

Reply:
In another direction, `ThreadFs` is a bit more consistent with `ProcessVm`.

> typically the name `XXXFs` indicates a specific type of filesystem itself.

But this also sounds reasonable. Theoretically, if we have procfs, maybe we can have threadfs, but that's certainly not what this type means.

Reply:
> In another direction, `ThreadFs` is a bit more consistent with `ProcessVm`.

`ThreadFs` and `ProcessVm` might have slight differences. `ProcessVm` basically have everything it needs to describe a "user space virtual memory". While `ThreadFs` basically contains `FsResolver` only, which is used for file path lookup in a filesystem, that's a small part of a `Fs`.

Reply:
I generally want to avoid `XxxInfo` as a type name, but on a case-by-case basis I cannot think of a better name. 

Since I agree that `ThreadFsInfo` might be a bit clearer than `ThreadFs`, I have renamed it as such.

Reply:
> I generally want to avoid XxxInfo as a type name

Agreed on that, so feel free to decide it yourself.

Reply:
> Since I agree that `ThreadFsInfo` might be a bit clearer than `ThreadFs`, I have renamed it as such.

Sorry. I forgot to run `git commit --amend`, so the changes got lost in the `git push`. My mistake.